modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/service/PluggableAuthenticationService.java

```diff
@@ -264,44 +175,25 @@ public String getSpecificAuthChainName(HttpServletRequest request) {
     }
 
     public List<NuxeoAuthenticationPlugin> getPluginChain() {
-        List<NuxeoAuthenticationPlugin> result = new ArrayList<>();
-
-        for (String pluginName : authChain) {
-            if (authenticatorsDescriptors.containsKey(pluginName)
-                    && authenticatorsDescriptors.get(pluginName).getEnabled()) {
-                if (authenticators.containsKey(pluginName)) {
-                    result.add(authenticators.get(pluginName));
-                }
-            }
-        }
-        return result;
+        return Collections.unmodifiableList(authPluginChain);
     }
 
     public NuxeoAuthenticationPlugin getPlugin(String pluginName) {
-        if (authenticatorsDescriptors.containsKey(pluginName)
-                && authenticatorsDescriptors.get(pluginName).getEnabled()) {
-            if (authenticators.containsKey(pluginName)) {
-                return authenticators.get(pluginName);
-            }
-        }
-        return null;
+        return authenticators.get(pluginName);
     }
 
     public AuthenticationPluginDescriptor getDescriptor(String pluginName) {
-        if (authenticatorsDescriptors.containsKey(pluginName)) {
-            return authenticatorsDescriptors.get(pluginName);
-        } else {
-            log.error("Plugin " + pluginName + " not registered or not created");
-            return null;
-        }
+        return this.<AuthenticationPluginDescriptor> getRegistryContribution(EP_AUTHENTICATOR, pluginName)
+                   .orElseGet(() -> {
+                       log.error("Plugin '{}' not registered or not created", pluginName);
+                       return null;
+                   });
     }
 
     public void invalidateSession(ServletRequest request) {
         boolean done = false;
-        if (!sessionManagers.isEmpty()) {
-            Iterator<NuxeoAuthenticationSessionManager> it = sessionManagers.values().iterator();
-            while (it.hasNext() && !(done = it.next().invalidateSession(request))) {
-            }
+        Iterator<NuxeoAuthenticationSessionManager> it = sessionManagers.values().iterator();
+        while (it.hasNext() && !(done = it.next().invalidateSession(request))) {
         }
```

Comment:
FYI this would work too I think:

    sessionManagers.values().stream().takeWhile(sm -> sm.invalidateSession(request)).count();

But I'm not actually suggesting we do this as it's probably not that clear to the reader.

Reply:
sonarlint was not happy with the empty {} indeed... but I did give up trying to find another way :)